### PR TITLE
feat: add assistant bot templates

### PR DESCRIPTION
**Pull request by swatDong**
https://msazure.visualstudio.com/Microsoft%20Teams%20Extensibility/_workitems/edit/25757244/
E2E Test: N/A

Able to create *AI Assistant Bot*
image

And can run in Teams
ai assistant bot in Teams

### CODE REVIEW

File: templates/js/ai-assistant-bot/README.md.tpl [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+>
+> **Note**
+>
+> Teams AI Library is currently in preview version.
+
+### Create your own OpenAI Assistant
+
+Before running or debugging your bot, please follow these steps to setup your own OpenAI Assistant.
+
+**If you haven't setup any Assistant yet**
```
Comment (kimizhu) [inline templates/js/ai-assistant-bot/README.md.tpl:31]: I suggest saying something about create.ts to show we provide the script to create assistant, customize if needed.

**Review by kuojianlu — APPROVED** [review]

File: packages/fx-core/resource/package.nls.json [hunk 2]
```diff
@@ -309,8 +309,10 @@
   "core.createProjectQuestion.openAiPluginManifest.validationError.missingApiUrl": "Missing URL in \"%s\".",
   "core.createProjectQuestion.openAiPluginManifest.validationError.authNotSupported": "Auth type is not supported. Supported auth type: \"%s\".",
   "core.createProjectQuestion.meArchitecture.title": "Architecture of Search Based Message Extension",
+  "core.aiAssistantBotOption.label": "AI Assistant Bot",
+  "core.aiAssistantBotOption.detail": "A custom AI assistant bot in Teams using Teams AI library and OpenAI Assistants API",
   "core.aiBotOption.label": "AI Chat Bot",
-  "core.aiBotOption.detail": "A chat bot using Teams AI library",
```
Comment (kimizhu) [inline packages/fx-core/resource/package.nls.json:313]: why remove "chat"?

File: templates/js/ai-assistant-bot/README.md.tpl (hunk 1 above)
Comment (swatDong) [reply templates/js/ai-assistant-bot/README.md.tpl:31]: Added:

> This app template provides script `src/creator.ts` to help create assistant. You can customize the instructions and settings there.

File: packages/fx-core/resource/package.nls.json (hunk 2 above)
Comment (swatDong) [reply packages/fx-core/resource/package.nls.json:313]: Added back

**Review by kuojianlu — APPROVED** [review]

**Review by kimizhu — DISMISSED** [review]

**Review by kuojianlu — APPROVED** [review]

File: templates/js/ai-assistant-bot/README.md.tpl [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+>
+> Teams AI Library is currently in preview version.
+
+### Create your own OpenAI Assistant
+
+Before running or debugging your bot, please follow these steps to setup your own [OpenAI Assistant](https://platform.openai.com/docs/assistants/overview).
+
+**If you haven't setup any Assistant yet**
+
+> This app template provides script `src/creator.js` to help create assistant. You can customize the instructions and settings there.
```
Comment (kimizhu) [inline templates/js/ai-assistant-bot/README.md.tpl:33]: Is it better to change to 
> This app template provides script `src/creator.js` to help create assistant. You can change the instructions and settings in the script to customize the assistant.
> 
> After creation, you can change and manage your assistants on [OpenAI](https://platform.openai.com/assistants).

Comment (swatDong) [reply templates/js/ai-assistant-bot/README.md.tpl]: Updated

**Review by sffamily — APPROVED** [review]

**Review by jayzhang — APPROVED** [review]

**Review by kimizhu — APPROVED** [review]

**Pull request merged**